### PR TITLE
[Feature](function) Support xxhash function like murmur hash function

### LINKED ISSUE #31138: [Feature] Support xxhash function like murmur hash funcion
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/doris/issues?q=is%3Aissue) and found no similar issues.

### Description

image

Support xxhash like murmur hash to speed up hash compute

image

### Use case

_No response_

### Related issues

_No response_

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by sjyango**
## Proposed changes

Issue Number: close #31138

Support `xxhash_32` and `xxhash_64` fucntion. 
```SQL
mysql> select xxhash_32('hello', 'world', '!');
+----------------------------------+
| xxhash_32('hello', 'world', '!') |
+----------------------------------+
|                        352087701 |
+----------------------------------+
1 row in set (0.06 sec)

mysql> select xxhash_32('hello', 'world', NULL);
+-----------------------------------+
| xxhash_32('hello', 'world', NULL) |
+-----------------------------------+
|                              NULL |
+-----------------------------------+
1 row in set (0.01 sec)

mysql> select xxhash_64('hello');
+----------------------+
| xxhash_64('hello')   |
+----------------------+
| -7685981735718036227 |
+----------------------+
1 row in set (0.07 sec)

mysql> select xxhash_64('hello', 'world');
+-----------------------------+
| xxhash_64('hello', 'world') |
+-----------------------------+
|         7001965798170371843 |
+-----------------------------+
1 row in set (0.06 sec)

mysql> select xxhash_64('hello', NULL);
+--------------------------+
| xxhash_64('hello', NULL) |
+--------------------------+
|                     NULL |
+--------------------------+
1 row in set (0.01 sec)

mysql> select xxhash_64('hello', 'world', '!');
+----------------------------------+
| xxhash_64('hello', 'world', '!') |
+----------------------------------+
|              6796829678999971400 |
+----------------------------------+
1 row in set (0.06 sec)

mysql> select xxhash_64('hello', 'world', NULL);
+-----------------------------------+
| xxhash_64('hello', 'world', NULL) |
+-----------------------------------+
|                              NULL |
+-----------------------------------+
1 row in set (0.01 sec)
```

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by sjyango** [comment]
run buildall

**Comment by sjyango** [comment]
run buildall

File: be/src/vec/functions/function_hash.cpp [hunk 1]
```diff
@@ -149,11 +149,108 @@ struct MurmurHash3Impl {
...
+
+    static Status combine_apply(const IDataType* type, const IColumn* column,
+                                size_t input_rows_count, IColumn& icolumn) {
+        return execute<false>(type, column, input_rows_count, icolumn);
+    }
+
+    template <bool first>
+    static Status execute(const IDataType* type, const IColumn* column, size_t input_rows_count,
+                          IColumn& col_to) {
+        auto& col_to_data = assert_cast<ColumnVector<ReturnType>&>(col_to).get_data();
```
Comment (HappenLee) [inline be/src/vec/functions/function_hash.cpp:179]: if (first)
   col_to.insert_many_defaults(col_to)

File: be/src/vec/functions/function_hash.cpp [hunk 2]
```diff
@@ -41,21 +41,84 @@ namespace doris::vectorized {
...
-struct MurmurHash3ImplName<Int64> {
-    static constexpr auto name = "murmur_hash3_64";
+    template <bool first>
+    static Status execute(const IDataType* type, const IColumn* column, size_t input_rows_count,
+                          IColumn& col_to) {
+        auto& to_column = assert_cast<ColumnVector<ReturnType>&>(col_to);
+        if constexpr (first) {
+            if constexpr (std::is_same_v<ReturnType, Int32>) {
+                for (int i = 0; i < input_rows_count; ++i) {
+                    to_column.insert_value(static_cast<Int32>(HashUtil::MURMUR3_32_SEED));
```
Comment (HappenLee) [inline be/src/vec/functions/function_hash.cpp:71]: do not use insert_value
resize + (memset / std::fill) to do the work

**Comment by sjyango** [comment]
run buildall

**Comment by sjyango** [comment]
run buildall

**Review by HappenLee — APPROVED** [review]
LGTM

**Review by Gabriel39 — APPROVED** [review]

**Pull request merged**